### PR TITLE
MMIs and pAIs can now be teleported properly.

### LINKED ISSUE #126: Invisible MMI
Reported by locodingo_05@yahoo.com, Jan 5, 2013
Teleported a MMI with a brain in it into robotics after the cyborg was blown. The MMI was on the mining asteroid and I when I teleported it, the MMI disappeared, but the player was on the tile and still able to speak to other players.

Revision 5461

TG-Sibyl 1

Problem Description: I expected to teleport the MMI into robotics from the asteroid without any issues. The problem I see is that if someone tried to teleport a MMI, that MMI will go invisible and not be seen by anyone and would not be helped back into a body.

 
1)Mining cyborg blown on mining asteroid.
2)Click "Get Key" and enter player's ckey while in robotics.

Possibly related stuff: I don't believe there was anything related to it, but the game mode was Changeling.

Anything else?: I'm not sure if this is a one time occurence, but I believe it should be looked into at the earliest convenience.

Jan 11, 2013 
Project Member 1 sieve32
Did you teleport the mob or the object? Brains are actually 2 separate things, the /obj/item/brain that people can pick up and whatnot, and the /mob/living/carbon/brain that exists inside the brain obj (Which is invisible). The MMI just being a modifier of the brain.
Jan 27, 2013 
Project Member 2 petethegoat
Pretty sure that would have just teleported the mob, not the brain obj.
Feb 21, 2013 
Project Member 3 giacomand
Pretty much. I don't think there's a good way to fix this other than to just tell admins to move the brain instead of the mob inside the brain.
Mar 9, 2013 
Project Member 4 petethegoat
(No comment was entered for this change.)
Status: Accepted
Labels: -Type-Task -Priority-High Type-Defect Priority-Medium

### PR DESCRIPTION
**Pull request by theOperand**
The admin get/jump procs have been revamped to use admin_forcemove(), which first sets various vars (pass_flags, incorporeal_move, density), Move()s the mob to the new loc, and then resets these vars to their original value.

It's a huge fucking hack, but neccesary for this to work.

Fixed #126 

Also, I fucked up the original commit message, so welp.

### CODE REVIEW

File: code/modules/admin/verbs/adminjump.dm [hunk 1]
```diff
@@ -126,8 +129,25 @@
 		return
 	var/area/A = input(usr, "Pick an area.", "Pick an area") in return_sorted_areas()
 	if(A)
-		M.loc = pick(get_area_turfs(A))
+		admin_forcemove(pick(get_area_turfs(A)))
```
Comment (phil235) [inline code/modules/admin/verbs/adminjump.dm]: You forgot to add the mob argument here.

Comment (Miauw62) [inline code/modules/admin/verbs/adminjump.dm]: welp

**Comment by Miauw62** [comment]
Alright, done.
Sorry @DeanTwo, but I forgot to pull from origin so it was easier to discard your commit. Thanks for the effort anyway.

**Comment by Deantwo** [comment]
Hehe, it is ok.
I just couldn't fall aslep last night.

**Comment by Miauw62** [comment]
thanks for causing conflicts, hg.
now the branch is completely fucked, so i guess this bug will go unfixed.

**Pull request closed without merging**